### PR TITLE
Successfully build libmarpa-sys

### LINKED ISSUE #5: Make the autotools build of libmarpa-sys suck less
I apparently broke things for everyone *not* using autotools 1.16 when I ran `autoreconf -i` so that I could run `cargo publish` from my computer. Might need to revert that change, but it still won't solve my problem of `cargo package` failing because it tries to reconfigure libmarpa during `make install` for some reason.

More exploration needed.

### PR DESCRIPTION
**Pull request by dginev**
Fixes #5 , at least to the point of building fully again.

Two tidbits: 
 * Now that the `libmarpa-sys` crate is attached in the main marpa repo, the dependency should be based on a path attribute, rather than on a version number
 * there was a missing `crate::` annotation in the test.rs of libmarpa-sys, needed for the 2018 edition

With those in place, I did a `cargo clean` and rebuilt from the updated `autoreconf -i` commit at bdd4846cd722e81f6d0874a3f17dd78acd19adb2 , and everything worked.

### CODE REVIEW

**Comment by dginev** [comment]
Will be interesting to see if Travis-CI agrees with me here, may need another addition for different than v1.16 autotools, which is what I used locally.

**Comment by dginev** [comment]
Ugh, shouldn't have resolved the conflict with master via a merge, this is a mess now... Will try to rebase on it locally and force push here...

**Comment by dginev** [comment]
Looks like if I could only propagate the `-fPIC` flag to the right compile call, Travis will be happy with a much older version of autotools.

**Comment by dginev** [comment]
I feel... uncertain?... about the fixes I ended up with here, but I made travis happy and my local build works nicely. 

The one very _very_ slippery bit I unearthed is that the Rust autotools package lets you set a `.cflag()` but - to the best of what I can glean from its code - never passes that flag onto the compiler call. Dropping that and directly extending `CFLAGS` worked as one would anticipate, and it's a relatively sane approach, at least a sane stopgap.

I feel a bit more guilty about dropping entirely the version flag and letting the OS pick its current `aclocal` executable though, since it may lead to very bizarre errors and is somewhat barbaric to the configure paradigm...

Make of it what you will, I hope at least it is a good brainstorm starter @jrobsonchase

File: marpa/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,10 @@ repository = "https://github.com/jrobsonchase/marpa"
...
 edition = "2018"
 
+[lib]
+crate-type = ["lib"]
+
 [dependencies]
 regex = "1.1.0"
 regex-syntax = "0.6.4"
-libmarpa-sys = "0.1.3"
+libmarpa-sys = {path="../libmarpa-sys"}
```
Comment (jrobsonchase) [inline marpa/Cargo.toml:17]: Whoops, could've sworn I converted this to a path dep! Maybe it got accidentally reverted or something.

Anyway, we should actually use *both*, that way it uses the path when developing locally, but can still be pushed to crates.io:

```toml
[dependencies.libmarpa-sys]
version = "0.1.3"
path = "../libmarpa-sys"
```

**Comment by jrobsonchase** [comment]
Ugh, so it builds now, but I end up with this when I try to package/publish:

```
error: failed to verify package tarball

Caused by:
  Source directory was modified by build.rs during cargo publish. Build scripts should not modify anything outside of OUT_DIR. Modified file: /data/home/jchase/src/github.com/jrobsonchase/marpa/target/package/libmarpa-sys-0.1.3/libmarpa-8.3.0/config.h.in
```

I hate autotools :confused:

**Comment by jrobsonchase** [comment]
So [this](https://github.com/jrobsonchase/marpa/commit/69efc509e5c95366a8cb45155a85875377d51bdb) builds for me locally, but completely bombs when I go to package/publish it:

```
configure: creating ./config.status
config.status: creating Makefile
config.status: creating LIB_VERSION
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
running: "make" "install"
CDPATH="${ZSH_VERSION+.}:" && cd /data/home/jchase/src/github.com/jrobsonchase/marpa/target/package/libmarpa-sys-0.1.3/libmarpa-8.3.0 && /bin/sh /data/home/jchase/src/github.com/jrobsonchase/marpa/target/package/libmarpa-sys-0.1.3/libmarpa-8.3.0/missing aclocal-1.15 -I m4

--- stderr
/data/home/jchase/src/github.com/jrobsonchase/marpa/target/package/libmarpa-sys-0.1.3/libmarpa-8.3.0/missing: line 81: aclocal-1.15: command not found
WARNING: 'aclocal-1.15' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <http://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <http://www.gnu.org/software/autoconf>
         <http://www.gnu.org/software/m4/>
         <http://www.perl.org/>
make: *** [Makefile:432: /data/home/jchase/src/github.com/jrobsonchase/marpa/target/package/libmarpa-sys-0.1.3/libmarpa-8.3.0/aclocal.m4] Error 127
thread 'main' panicked at '
command did not execute successfully, got: exit code: 2

build script failed, must exit now', /home/jchase/.cargo/registry/src/github.com-1ecc6299db9ec823/autotools-0.1.3/src/lib.rs:357:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:70
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:58
             at src/libstd/panicking.rs:200
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:215
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:478
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:385
   6: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:340
   7: autotools::fail
             at /home/jchase/.cargo/registry/src/github.com-1ecc6299db9ec823/autotools-0.1.3/src/lib.rs:357
   8: autotools::run
             at /home/jchase/.cargo/registry/src/github.com-1ecc6299db9ec823/autotools-0.1.3/src/lib.rs:345
   9: autotools::Config::build
             at /home/jchase/.cargo/registry/src/github.com-1ecc6299db9ec823/autotools-0.1.3/src/lib.rs:321
  10: build_script_build::main
             at ./build.rs:11
  11: std::rt::lang_start::{{closure}}
             at /rustc/bf669d1e3295bc688f71b8c91f48a6beaf895f67/src/libstd/rt.rs:64
  12: std::panicking::try::do_call
             at src/libstd/rt.rs:49
             at src/libstd/panicking.rs:297
  13: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:92
  14: std::rt::lang_start_internal
             at src/libstd/panicking.rs:276
             at src/libstd/panic.rs:388
             at src/libstd/rt.rs:48
  15: std::rt::lang_start
             at /rustc/bf669d1e3295bc688f71b8c91f48a6beaf895f67/src/libstd/rt.rs:64
  16: main
  17: __libc_start_main
  18: _start
```

At this point, I almost think it could be easier to update to a newer libmarpa version and hope it doesn't have the same autotools weirdness. I'm not sure why it's even running aclocal during make...

**Comment by dginev** [comment]
@jrobsonchase I don't mind at all updating to a new version -- especially if some part of the build ends up easier, though I don't think the libmarpa setup has changed significantly?

I am a bit curious why you decided to bring in the libmarpa-sys crate into the main marpa crate btw, especially after both were published in crates.io. That has confused me a bit with the recent changes in master.

**Comment by jrobsonchase** [comment]
> I am a bit curious why you decided to bring in the libmarpa-sys crate into the main marpa crate btw, especially after both were published in crates.io. That has confused me a bit with the recent changes in master.

I originally put this together before cargo workspaces were a thing. That's the only reason I can think of that I would've put them in separate crates, because I definitely wouldn't have made the same decision today. These days, it makes a lot more sense to keep crates that are as closely related as `-sys` bindings and the higher-level idiomatic interface in the same repository. See [openssl](https://github.com/sfackler/rust-openssl), [sodium](https://github.com/sodiumoxide/sodiumoxide), etc. as examples.

I'll take one more crack at getting it to build properly with the current version before giving up and pulling in the latest.

**Comment by dginev** [comment]
Cool, thanks for clarifying that, makes perfect sense. A good explanation to add to the release notes too, if anyone else gets curious :+1:

**Comment by dginev** [comment]
Closing PR to review your approach in #7

**Pull request closed without merging**